cpp/src/ray/runtime/abstract_ray_runtime.cc

```diff
@@ -129,8 +129,13 @@ std::vector<std::unique_ptr<::ray::TaskArg>> TransformArgs(
           memory_buffer, nullptr, std::vector<rpc::ObjectReference>()));
     } else {
       RAY_CHECK(arg.id);
-      ray_arg = absl::make_unique<ray::TaskArgByReference>(ObjectID::FromBinary(*arg.id),
-                                                           ray::rpc::Address{},
+      auto id = ObjectID::FromBinary(*arg.id);
+      auto owner_address = ray::rpc::Address{};
+      if (CoreWorkerProcess::IsInitialized()) {
```

Comment:
Is there any reason to check this?

Reply:
I think it is workaround for local mode. Maybe it ought to check if the config is native mode instead?

Reply:
That makes sense. Could you use `ConfigInternal::Instance().run_mode == RunMode::CLUSTER` instead?